### DISCUSSION TITLE
nominatim flatnode slow question

### DISCUSSION THREAD

**Discussion by ashley-mort** [osm-search/Nominatim#3058] [Q&A]
If I enable flatnode, my data load is super slow.  I'm working with the socal data from geofabrik now (southern California ~500MB .pbf).  I am using flatnode because my eventual goal is to try a full planet load.

My flatnode file is over 30GB right now and growing and has been running for over an hour.  I am kind of a newbie on this.  Any suggestions to what I'm doing wrong?

My command:
_nominatim import --osm-file /nominatim/flatnode/socal-latest.osm.pbf --threads 16_

My server has 32GB of RAM but HDD disk.

Any general advice or thoughts from someone with more experience than me would be appreciated!

**Comment by mtmail** [comment]
The flatnode file will grow to about 80GB regardless how big the input data is. It has advantages when you import very large countries (e.g. whole United States) or continents (Europe), but you don't need it for your data. The file is also accessed randomly throughout the import, so only makes sense on SDD/Nvme drives.

**Comment by nikitadtastewise** [comment]
@mtmail Can I run `nominatim api` without flatnode file after the import? flatnode file is needed only for future updates, isn't it?

**Reply by mtmail** [reply]
Flatfile is a faster replacemant for the planet_osm_node database table. If you import without flatfile then updates also work without, too.
